File: sample/python/MoveBodies.py

from cnoid.Util import *
from cnoid.Base import *
from cnoid.BodyPlugin import *
from numpy import *
import time

bodyItems = RootItem.instance().getDescendantItems(BodyItem)

for bodyItem in bodyItems:
    body = bodyItem.body()
    rootLink = body.rootLink()
    for i in range(20):
        rootLink.p += array([0, 0, 0.01])
        body.calcForwardKinematics()
        bodyItem.notifyKinematicStateChange()
        MessageView.instance().flush()
        time.sleep(0.01)

File: sample/lua/MoveBodies.lua
cnoid.require "Util"
cnoid.require "Base"
cnoid.require "Body"
cnoid.require "BodyPlugin"

bodyItems = cnoid.RootItem.instance():getDescendantItems(cnoid.BodyItem)

for i, bodyItem in ipairs(bodyItems) do
   body = bodyItem:body()
   rootLink = body:rootLink()
   for i=1,20 do
      rootLink:setTranslation(rootLink:translation() + cnoid.Vector3(0, 0, 0.01))
      body:calcForwardKinematics()
      bodyItem:notifyKinematicStateChange()
      cnoid.MessageView.instance():flush()
      cnoid.sleep(0.01)
   end
end
